src/chocolatey.resources/helpers/functions/Get-ChocolateyUnzip.ps1

```diff
@@ -91,32 +91,73 @@ param(
     $destination32 = $destination
   }
 
-  $exitCode = -1
-  $unzipOps = {
-    param($7zip, $destination, $fileFullPath, [ref]$exitCodeRef)
-    $params = "x -aoa -o`"$destination`" -y `"$fileFullPath`""
-    Write-Debug "Executing command ['$7zip' $params]"
-    $process = New-Object System.Diagnostics.Process
-    $process.StartInfo = new-object System.Diagnostics.ProcessStartInfo($7zip, $params)
-    $process.StartInfo.UseShellExecute = $false
-    $process.StartInfo.WindowStyle = [System.Diagnostics.ProcessWindowStyle]::Hidden
-
-    $process.Start() | Out-Null
-    $process.WaitForExit()
-    $processExitCode = $process.ExitCode
-    $process.Dispose()
-    Write-Debug "Command ['$7zip' $params] exited with `'$processExitCode`'."
-
-    $exitCodeRef.Value = $processExitCode
-  }
-
+  # Make structures used by .NET for starting the 7z process
+  
+  # 7z command line
+  $params = "x -aoa -o`"$destination`" -y `"$fileFullPath`""
+  $process = New-Object System.Diagnostics.Process
+  $process.StartInfo = new-object System.Diagnostics.ProcessStartInfo($7zip, $params)
+  # Required for stdout redirect
+  $process.StartInfo.UseShellExecute = $false
+  $process.StartInfo.WindowStyle = [System.Diagnostics.ProcessWindowStyle]::Hidden
+
+  Write-Debug "Executing command ['$7zip' $params]"
   if ($zipExtractLogFullPath) {
-    Write-Debug "wrapping 7za invocation with Write-FileUpdateLog"
-    Write-FileUpdateLog -logFilePath $zipExtractLogFullPath -locationToMonitor $destination -scriptToRun $unzipOps -argumentList $7zip,$destination32,$fileFullPath32,([ref]$exitCode)
+    # Redirect stdout for processing by choco
+    $process.StartInfo.RedirectStandardOutput = $true
+    [void] $process.Start()
+    
+    # Capture 7z's output into a StringBuilder and write it out in blocks, to improve I/O performance.
+    $sb = New-Object -TypeName "System.Text.StringBuilder";
+    
+    # Remember whether anything was written to the file
+    $written = $false
+    # Open the file once
+    $wStream = new-object IO.FileStream($zipExtractLogFullPath, [IO.FileAccess]::Write)
+    $sWriter = new-object IO.StreamWriter($wStream)
+    
+    # Read each line from 7z's stdout synchroneously (ReadLine blocks).
+    # Since stderr is not redirected, it gets automatically printed to the console, avoiding deadlocks.
+    while(($process.StandardOutput -ne $null) -and (($line = $process.StandardOutput.ReadLine()) -ne $null)) {
+      if($line.StartsWith("Extracting")) {
+        # This is a line indicating an extracted file
+        $file = $destination + "\" + $line.Substring(12)

```

Comment:
In 7z's output in the form "Extracting  path\of\file\in\zip", the file path begins at character 12.